app/src/main/java/org/oppia/app/topic/TopicFragmentPresenter.kt

```diff
@@ -87,4 +93,49 @@ class TopicFragmentPresenter @Inject constructor(
   private fun getTopicViewModel(): TopicViewModel {
     return viewModelProvider.getForFragment(fragment, TopicViewModel::class.java)
   }
+
+  private fun logTopicEvents(tab: TopicTab){
+    when(tab){
+      TopicTab.INFO -> logInfoFragmentEvent(topicId)
+      TopicTab.LESSONS -> logLessonsFragmentEvent(topicId)
+      TopicTab.PRACTICE -> logPracticeFragmentEvent(topicId)
+      TopicTab.REVISION -> logRevisionFragmentEvent(topicId)
+    }
+  }
+
+  private fun logInfoFragmentEvent(topicId: String){
+    analyticsController.logTransitionEvent(
+      fragment.requireActivity().applicationContext,
+      oppiaClock.getCurrentCalendar().timeInMillis,
+      EventLog.EventAction.OPEN_INFO_TAB,
+      analyticsController.createTopicContext(topicId)
+    )
+  }
+
+  private fun logLessonsFragmentEvent(topicId: String){
+    analyticsController.logTransitionEvent(
+      fragment.requireActivity().applicationContext,
+      oppiaClock.getCurrentCalendar().timeInMillis,
+      EventLog.EventAction.OPEN_LESSONS_TAB,
+      analyticsController.createTopicContext(topicId)
+    )
+  }
+
+  private fun logPracticeFragmentEvent(topicId: String){
```

Comment:
space before `{`, If you try doing `reformat code` from the menu toolbar such issues resolves automatically.

Reply:
Done.